packages/types/src/parser-options.ts

```diff
@@ -58,6 +78,7 @@ interface ParserOptions {
   tsconfigRootDir?: string;
   warnOnUnsupportedTypeScriptVersion?: boolean;
   moduleResolver?: string;
+  [additionalProperties: string]: ParserOptionsValue | undefined;
```

Comment:
You needn't add a specific type for the index signature because [typescript will strictly type the known keys even with an unbounded index signature](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgCoQM5mQbwFDKHJwBcyWUoA5gNwFEBGZIArgLYPR1HIJkMB7AQBsIcENyIBtANYQAnmQrUAumRYgZIAQHcJeAL548CASCzIAHmXQWAvLnqFSyAIwAaJ8ibIATJ54+ZABmTy8AEzIAFgCiCDIAcgTYwhgyMCgWCE8DGiA)

```suggestion
  [additionalProperties: string]: unknown;
```